apps/api/models.py

```diff
@@ -40,37 +48,61 @@ class Identification(models.Model):
     description = models.CharField(max_length=500, blank=True)
     number = models.PositiveIntegerField()
 
+    def __str__(self):
+        return str(self.student)
+
+    def __unicode__(self):
+        return str(self.student)
+
 
 class Lecture(models.Model):
     code = models.CharField(max_length=20)
     instructors = models.ManyToManyField("Instructor", blank=True)
+    event = models.ForeignKey("TAidEvent", null=True, blank=True)
     students = models.ManyToManyField("Student", blank=True)
 
+    def __str__(self):
+        return self.code
+
     def __unicode__(self):
         return self.code
 
 
 class Tutorial(models.Model):
     code = models.CharField(max_length=20)
-    ta = models.ManyToManyField("TeachingAssistant", blank=True)
+    teaching_assistant = models.ManyToManyField(
+        "TeachingAssistant", blank=True, verbose_name='Teaching Assistant(s)')
+    event = models.ForeignKey("TAidEvent", null=True, blank=True)
     students = models.ManyToManyField("Student", blank=True)
 
+    def __str__(self):
+        return self.code
+
     def __unicode__(self):
         return self.code
 
 
 class Practical(models.Model):
     code = models.CharField(max_length=20)
-    ta = models.ManyToManyField("TeachingAssistant", blank=True)
+    teaching_assistant = models.ManyToManyField(
+        "TeachingAssistant", blank=True, verbose_name='Teaching Assistant(s)')
+    event = models.ForeignKey("TAidEvent", null=True, blank=True)
     students = models.ManyToManyField("Student", blank=True)
 
+    def __str__(self):
+        return self.code
+
     def __unicode__(self):
         return self.code
 
 
 class Assignment(models.Model):
     name = models.CharField(max_length=254)
-    rubric_entries = models.ManyToManyField("Rubric", blank=True, related_name="rubric_entries")
+    rubric_entries = models.ManyToManyField(
```

Comment:
Is a many-to-many field required for rubrics? An assignment has many rubrics. Does a rubric have many assignments?